bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -268,6 +269,23 @@ public void run() {
                     logger.debug("Changed ThingType of Thing {} to {}. New ThingHandler is {}.",
                             thing.getUID().toString(), thing.getThingTypeUID(), thing.getHandler().toString());
                 }
+
+                private void waitForRunningHandlerRegistrations(ThingUID thingUID) {
+                    for (int i = 0; i < 10 * 10; i++) {
```

Comment:
yes sure - waiting for 100ms * 10 = 1sec * 10 = 10sec. 
should have helped readability. Obviously failed :-D

To be honest, I don't like the whole "locking" mechanism that works by simply ignoring a second attempt to register a handler at the same time. I'd rather spend some time on reworking the hole notification chain in order to not run into such cycles from the beginning... Anyway, this change is effective and relieves the immediate pain.